src/nupic/math/ArrayAlgo.hpp

```diff
@@ -24,4952 +24,6021 @@
  * Algorithms on arrays, dense or sparse. Contains speed-optimized asm
  * code that uses SIMD instructions provided by SSE in x86, only for darwin86
  * so far.
+ *
+ * Notes:
+ *	bind2nd() is depriciated in C++11 and removed in C++17 so
+ *            bind2nd is replaced with C++11 bind(). At some point we may
+ *            want to consider using C++14 generic lambda which they say is faster.
+ *  Compiles under Windows C++17 and Linux C++11.  Asm no used in Windows.
```

Comment:
the comment and change to bind() is :+1: 
but this file is probably a merge conflict! Too many changes here, 
please see the original 
https://github.com/htm-community/nupic.cpp/blob/master/src/nupic/math/ArrayAlgo.hpp 
And apply your change to that. 
I think you've re-added the ASM stuff here (which might be the reason why CI builds are failing) 
((also, this removed ASM is another thing to add to the "Changes", although it's not API(?) )

Reply:
This is a tough one.  I am trying to find something that will compile under C++11 and C++17.  I ended up getting really messed up such that nothing compiled.  So I went back to the original...perhaps I went back too far.  It now compiles.  I don't think the ASM stuff is all that bad. This was not the problem with CI.  It will not work on Windows but in those cases where it matters it does make a big difference in speed.   But I will work on this some more and reduce the conflicts.

Reply:
> compile under C++11 and C++17 

for other PR, but.. you've convinced me to go c++17 way (but there were some problems with python you've later discovered?), so if there are problems between ++11/17, should we move to the middleman c++14 as a sub-goal? 

> where it matters it does make a big difference in speed 

Do you have a test for this? Because I was deciding just that, and in my evaluation the plain c++(with compiler optimizations) made no difference compared to the ASM - therefore I went on pruning all of the ASM. 

Anyway, my comment was that you got messed in rebasing from your other branch to the newer base, if there's not too much changes, I'd suggest go with the current file from this repo and rewrite your changes there manually.

Reply:
Ok, re-did this file.  Went back to nupic.cpp master, and re-made just my changes. .. not many.  Compiles on C++11 and C++17.